Modules/_io/winconsoleio.c

```diff
@@ -133,6 +133,21 @@ char _PyIO_get_console_type(PyObject *path_or_fd) {
     return m;
 }
 
+static HANDLE py_get_osfhandle(int fd)
```

Comment:
Do you wish me to:
1. Do this entire refactor right here in this PR.
2. Add the functions and only use them in winconsoleio.c and _testconsole.c which I touched in this PR and leave the rest of the modules to a separate PR.
3. Leave this to a separate PR entirely.